website/docs/cloud-docs/registry/publish-providers.mdx

```diff
@@ -20,72 +20,196 @@ Users must be members of an organization to access its registry and private prov
+### Create the provider
+
+Create a file named `provider.json` with the following contents. Replace `PROVIDER_NAME` with the name of your provider and replace `ORG_NAME` with the name of your organization.
+
+```json
+{
+  "data": {
+    "type": "registry-providers",
+    "attributes": {
+      "name": "PROVIDER_NAME",
+      "namespace": "ORG_NAME",
+      "registry-name": "private"
+    }
+  }
+}
+```
+
+Use the [Create a Provider endpoint](/terraform/cloud-docs/api-docs/private-registry/providers#create-a-provider) to create the provider in Terraform Cloud. Replace `TOKEN` in the `Authorization` header with your Terraform Cloud API token and replace `ORG_NAME` with the name of your organization.
+
+```shell-session
+$ curl \
+  --header "Authorization: Bearer TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request POST \
+  --data @provider.json \
+  https://app.terraform.io/api/v2/organizations/ORG_NAME/registry-providers
+```
+
+The provider is now available in your organization’s Terraform Cloud private registry, but consumers cannot use it until you add a version and a platform. To create them, you will will need the provider binaries, public GPG signing key, `SHA256SUMS` file, and `SHA256SUMS.sig` file from at least one release.
+
+
+### Add your public key
+
+-> **NOTE**: If you are uploading a public provider into your private registry, skip this step and instead use [HashiCorp's public PGP key](https://www.hashicorp.com/.well-known/pgp-key.txt). To retrieve the `key-id` of HashiCorp's public PGP key, TODO
+
+Create a file named `key.json` with the following contents. Replace `ORG_NAME` with the name of your organization and supply your public key in the `ascii-armor` field.
+
+```hcl
+{
+  "data": {
+    "type": "gpg-keys",
+    "attributes": {
+      "namespace": "ORG_NAME",
+      "ascii-armor": "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINB...=txfz\n-----END PGP PUBLIC KEY BLOCK-----\n"
+    }  }
+}
+```
+
+Use the [Add a GPG key endpoint](/terraform/cloud-docs/api-docs/private-registry/gpg-keys#add-a-gpg-key) to add the public key that matches the signing key for the release. Replace `TOKEN` in the `Authorization` header with your Terraform Cloud API token.
+
+```shell-session
+$ curl \
+  --header "Authorization: Bearer TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request POST \
+  --data @key.json \
+  https://app.terraform.io/api/registry/private/v2/gpg-keys
+```
+
+The response contains a `key-id` that you will use to create a provider version.
+
+```json
+"key-id": "34365D9472D7468F"
+```
+
+### Create a version
+
+Create a file named `version.json` with the following contents. Replace the value of the `version` field with the version of your provider, and replace the `key-id` field with the id of the GPG key that you created in the [Add your public key](#add-your-public-key) step.
```

Comment:
I think it makes sense to _also_ put a note here that says "if you're uploading a public provider, then use key-id 34365D9472D7468F"